### PR TITLE
Fix -sigopt option with "nonce-type" in openssl dgst

### LINKED ISSUE #25012: -sigopt option doesn't work with "nonce-type" in openssl dgst
Using the `nonce-type:1` as an option to the `openssl dgst` command doesn't work.

Tried 7fc3a59f3f62dc06 (current master) and 3.2.2.

```
openssl genpkey -out key.pem -algorithm EC -pkeyopt ec_paramgen_curve:P-256
echo message to sign > message.txt
openssl dgst -sha256 -sign key.pem -out message.sig1 -sigopt nonce-type:1 message.txt
```
fails with
```
Signature parameter error "nonce-type:1"
40E74E779F7F0000:error:03000093:digital envelope routines:default_fixup_args:command not supported:crypto/evp/ctrl_params_translate.c:578:[action:2, state:4] name=nonce-type, value=1
```

### PR DESCRIPTION
**Pull request by mattcaswell**
We apply two fixes:

1) Don't restrict the ECDSA settable ctx params unnecessarily

We were being a bit overzealous about what settable_ctx_params are reported for ECDSA and weren't always getting it right. Just report all the possible ones. We have checks anyway if you try and set things that aren't appropriate for a given situation.

2) Improve the error reporting if we end up sending an empty digest for deterministic ECDSA

Fixes #25012

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: crypto/ec/ecdsa_ossl.c [hunk 1]
```diff
@@ -106,6 +106,10 @@ int ossl_ecdsa_deterministic_sign(const unsigned char *dgst, int dlen,
         ERR_raise(ERR_LIB_EC, ERR_R_PASSED_NULL_PARAMETER);
         return 0;
     }
+    if (digestname == NULL) {
+        ERR_raise(ERR_LIB_EC, EC_R_INVALID_DIGEST);
```
Comment (beldmit) [inline crypto/ec/ecdsa_ossl.c:110]: Don't we have any error with "not specified" semantics? Or add the digest name as extra info?

**Comment by beldmit** [comment]
Doesn't it make sense adding the case from #25012 as a test?

File: crypto/ec/ecdsa_ossl.c (hunk 1 above)
Comment (InfoHunter) [reply crypto/ec/ecdsa_ossl.c:110]: The term "invalid digest" includes the case that no digest has been specified, so there is not a need for another error.

**Comment by mattcaswell** [comment]
> Doesn't it make sense adding the case from https://github.com/openssl/openssl/issues/25012 as a test?

Done. Please take another look.

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by mattcaswell** [comment]
@t8m please reconfirm

**Review by t8m — APPROVED** [review]

**Review by shahsb — APPROVED** [review]
LGTM. Thanks!

**Comment by t8m** [comment]
Merged to the master, 3.4, 3.3 and 3.2 branches. Thank you.

**Pull request closed without merging**